pylint/checkers/unicode.py

```diff
@@ -0,0 +1,538 @@
+                "of this issue. And "
+                "https://softwareengineering.stackexchange.com/questions/102205/should-utf-16-be-considered-harmful "
+                "for some possible problems when using UTF-16 for instance."
+            ),
+        ),
+        "E2502": (
+            (
+                "Contains control characters that can permit obfuscated code "
+                "executed differently than displayed"
+            ),
+            "bidirectional-unicode",
+            (
+                "bidirectional unicode are typically not displayed characters required "
+                "to display right-to-left (RTL) script "
+                "(i.e. Chinese, Japanese, Arabic, Hebrew, ...) correctly. "
+                "So can you trust this code? "
+                "Are you sure it displayed correctly in all editors? "
+                "If you did not write it or your language is not RTL,"
+                " remove the special characters, as they could be used to trick you into executing code, "
+                "that does something else than what it looks like.\n"
+                "More Information:\n"
+                "https://en.wikipedia.org/wiki/Bidirectional_text\n"
+                "https://trojansource.codes/"
+            ),
+        ),
+        "C2503": (
+            "PEP8 recommends UTF-8 as encoding for Python files",
+            "bad-file-encoding",
+            (
+                "PEP8 recommends UTF-8 default encoding for Python files. See "
+                "https://www.python.org/dev/peps/pep-0008/#id20"
+            ),
+        ),
+        **{
+            bad_char.code: (
+                bad_char.description(),
+                bad_char.human_code(),
+                bad_char.help_text,
+            )
+            for bad_char in BAD_CHARS
+        },
+    }
+
+    @staticmethod
+    def _is_invalid_codec(codec: str) -> bool:
+        return codec.startswith("utf-16") or codec.startswith("utf-32")
+
+    @staticmethod
+    def _is_unicode(codec: str) -> bool:
+        return codec.startswith("utf")
+
+    @classmethod
+    def _find_line_matches(cls, line: bytes, codec: str) -> Dict[int, _BadChar]:
+        """Find all matches of BAD_CHARS within line
+
+        Args:
+            line: the input
+            codec: that will be used to convert line/or search string into
+
+        Return:
+            A dictionary with the column offset and the BadASCIIChar
+        """
+        # We try to decode in Unicode to get the correct column offset
+        # if we would use bytes, it could be off because UTF-8 has no fixed length
+        try:
+            line_search = line.decode(codec, errors="strict")
+            search_dict = BAD_ASCII_SEARCH_DICT
+            # If everything is correctly decoded, the BOM will be removed automatically
+            return _map_positions_to_result(line_search, search_dict, "\n")
+        except UnicodeDecodeError:
```

Comment:
Could you add a comment explaining why this isn't covered currently

Reply:
I don't understand, if I debug `test_find_bad_chars` this is covered.
Or you mean why we need it at all?

Reply:
That's interesting! According to `coveralls` L400-412 isn't covered. Could this be due to the tests running on Linux? See: https://coveralls.io/builds/45561927/source?filename=pylint%2Fcheckers%2Funicode.py#L400

Reply:
If there is line that can't be decoded but consists only of a comment, python silently ignores the decode error.
So our linter would crash even with a valid python file.

Reply:
Can see the coverage:

![image](https://user-images.githubusercontent.com/7687556/149134149-933bd914-1755-48a7-9ade-aa8ff5d5a8bd.png)

I run the tests on linux myself.

Reply:
You need to create an account on their website and login to see it on a per-file basis. Something to do with them making calls to the Github servers and getting rate-limited if they did it from their own account instead of yours.

Reply:
LoL, the test have been named wrong. As they did  contained `test`, they weren't executed.
Actually one test even does not pass OMG, what a fail...

Good catch

Reply:
Oh wow. I keep forgetting to check file names when doing reviews. Good that we checked this indeed!

Reply:
I should have made a not also, because given the amount of test written, it was quite strange that I only had a 88.05% coverage....

Running locally I get 100% now (only using the unittests). 
If this isn't reflected by coveralls we have a problem.
Resolve this once coverall shows 100% coverage for unicode.py

Reply:
Thanks for the login note.
Coverage is now 100%. Finally.